Move show password checkbox under password field

sanjaysiddhanti opened the pull request:
@di 

This PR implements https://github.com/pypa/warehouse/issues/6722. 

Screenshot before:

![Screen Shot 2020-06-01 at 10 47 17 PM](https://user-images.githubusercontent.com/3238190/83484884-c2c62980-a45a-11ea-9da1-27f16b9237fb.png)

Screenshot after:

![Screen Shot 2020-06-01 at 10 50 33 PM](https://user-images.githubusercontent.com/3238190/83484904-cce82800-a45a-11ea-9509-2ac14e6dff4d.png)

Let me know if you'd like any additional changes.

Fixes #6722.

sanjaysiddhanti commented:
I could also move "show password" down to be horizontally aligned with "Log in" and "forgot password." I wasn't sure where exactly you wanted it. This does make password one tab away from username as the ticket requests.

di commented:
> I could also move "show password" down to be horizontally aligned with "Log in" and "forgot password."

Yes, please do that! I think that can be done with the `split-layout` class. We should also remove the `split-layout` class where it's no longer being used, like https://github.com/pypa/warehouse/issues/6722#issuecomment-538182534 describes.

sanjaysiddhanti commented:
Ok, I removed the `split-layout` where it is no longer used. When I apply the rest of the diff from https://github.com/pypa/warehouse/issues/6722#issuecomment-538182534, it basically looks like my screenshot above, without horizontal alignment.

How does d845979 look? I did the following:

- Moved "show password" into the form group that has the log in and forgot password buttons
- This broke the "show password" functionality. Looks like clicking on "show password" triggers some JS code, but it needs to be in a block that connects it to the Password Controller. So, I moved the entire form group into the `<div data-controller="password" class="form-group">` block. Is there a better solution?

New screenshot below

![image](https://user-images.githubusercontent.com/3238190/83576706-2e55d880-a4e7-11ea-8a9f-6bcf8026fad5.png)

di commented on warehouse/templates/accounts/login.html:
```diff
@@ -86,16 +82,19 @@ <h1 class="page-title">{% trans title=title %}Log in to {{ title }}{% endtrans %
+              <div>
+                <input type="submit" value="{% trans %}Log in{% endtrans %}" class="button button--primary">
+              </div>
+              <label for="show-password">
+                <input data-action="change->password#togglePasswords" data-target="password.showPassword"
+                  id="show-password" type="checkbox">&nbsp;{% trans %}Show password{% endtrans %}
+              </label>
+              <span>
+                <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
+              </span>
```
I _think_ wrapping these in a `div` will cause them to both be moved to the right side, but stacked on top of each other

```suggestion
              <div>
                <label for="show-password">
                  <input data-action="change->password#togglePasswords" data-target="password.showPassword"
                    id="show-password" type="checkbox">&nbsp;{% trans %}Show password{% endtrans %}
                </label>
                <span>
                  <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
                </span>
              </div>
```

sanjaysiddhanti commented:
>  will cause them to both be moved to the right side, but stacked on top of each other

Ah, I didn't realize that was the goal. Your suggested diff doesn't quite do it, since `label` and `span` are both inline (see below).

<img width="441" alt="Screen Shot 2020-06-02 at 5 56 23 PM" src="https://user-images.githubusercontent.com/3238190/83584060-de810c80-a4fa-11ea-8fbc-1c611324c733.png">

But when I wrap the `label` in a `div` in b98f393, I think it looks right

<img width="392" alt="Screen Shot 2020-06-02 at 5 58 04 PM" src="https://user-images.githubusercontent.com/3238190/83584113-06707000-a4fb-11ea-89b2-2303219304cd.png">

sanjaysiddhanti commented:
@nlhkabu No rush, but just a friendly ping on this one in case it got lost.

sanjaysiddhanti commented on warehouse/templates/accounts/login.html:
```diff
@@ -86,12 +82,15 @@ <h1 class="page-title">{% trans title=title %}Log in to {{ title }}{% endtrans %
-            <div>
-              <input type="submit" value="{% trans %}Log in{% endtrans %}" class="button button--primary">
+          <div class="form-group">
+            <div class="split-layout margin-top--large margin-bottom--large">
```
removed `split-layout--table`, as it causes elements to be vertically centered in the container, whereas Nicole's screenshot shows "Show password" aligned to the top of the div

sanjaysiddhanti commented:
Thanks for the feedback! Here is an updated version:

![image](https://user-images.githubusercontent.com/3238190/84357194-1da60200-ab7a-11ea-80b8-cd3e8fca333c.png)

The only difference I see is that Nicole's version has double the margin between "Log in" and "Show password." I tried to achieve this with the following, but the margins were getting collapsed with the sibling element:

```
            <span class="display-block margin-top--large">
              <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
            </span>
```
I also tried wrapping the `span` in a `div class="margin-top--large"` 

@di did you have an idea of how to get this extra margin and does it matter to you?

patelneel55 commented:
Since there's now a new feature request regarding a "Keep me signed in" button at #8033, would it be beneficial to switch to a :eye: icon, as mentioned by @bittner at [#6722](https://github.com/pypa/warehouse/issues/6722#issuecomment-613121450), to prevent a clutter of text fields under the password field?

Any thoughts? @di @sanjaysiddhanti

bittner commented:
Nice! Adding an :eye: icon can maybe done easily now. Let's see. :smiley_cat: 

Thanks for this improvement so far! :1st_place_medal:

bittner commented:
Wow! This change is already rolled out. Nice!

Is the pipeline that does this public? It doesn't seem to be the [GitHub Actions](https://github.com/pypa/warehouse/actions) of this repository. :thinking:

brainwane commented:
Hi @sanjaysiddhanti! Congrats on landing your first pull request in PyPA projects. Could I ask you to take a look at #5222 and comment there if you can work on it?
